projects/packages/assets/src/class-assets.php

```diff
@@ -317,7 +330,7 @@ public static function normalize_path( $path ) {
 	 *  - `css_path`:         (string|null) `.css` to load. Default is to base it on `$path`.
 	 *  - `dependencies`:     (string[]) Additional script dependencies to queue.
 	 *  - `enqueue`:          (bool) Set true to enqueue the script immediately.
-	 *  - `in_footer`:        (bool) Set true to register script for the footer.
+	 *  - `in_footer`:        (bool|array) Pass an array of args to leverage script_strategy since WordPress 6.4. Set true to register script for the footer.
```

Comment:
>We're already taking an array of parameters, let's either use the existing 'async' parameter or deprecate that and create a new 'strategy' parameter that takes 'defer' or 'async'

I would suggest matching the core API signature, so accept a `strategy`. in_footer is currently still supported, although it may be deprecated in core at some point in the future.

> (or explicitly null/false/empty-string, which core seems to have missed documenting).

I believe because the intention is  this might be extended to support additional strategies (even by plugins), for example `worker`.

Reply:
I switched to using a new `strategy` option for the array in https://github.com/Automattic/jetpack/pull/34072/commits/e341de8c39a9bbc4d351bd6d8f7db6bf34aa06d6, leaving `in_footer` unchanged. I also extended support to include `async` in https://github.com/Automattic/jetpack/pull/34072/commits/733270b997e2d4a13fed55a0b5c3a1b2afb2b844.

I noticed the function `add_async_script` (and `script_add_async`) actually adds `defer` to the script tags, is that correct? and you don't use the `async` attribute at all, right? 

Would you be open to renaming these function(s)? As is, they are confusing - should you want to use `async` and `defer` they become problematic!

Reply:
As I suggested [earlier](https://github.com/Automattic/jetpack/pull/34072#pullrequestreview-1727426868), we should probably deprecate those functions. Even if we renamed them we'd still need to keep the old confusing names around until the next major version bump, and there doesn't seem to be much point in having new correctly-named versions now that WordPress 6.3 includes the same functionality.

Reply:
> As I suggested earlier, we should probably deprecate those functions. 

Ok, not sure how to do that in Jetpack, I'll see if I can find some examples.

Reply:
Step one is to add an `@deprecated` tag in the phpdoc.

Step two would be to add a call to WordPress's `_deprecated_function()` function. But before doing this we'd want to clean up any existing uses within this repository. For `add_async_script` it looks like there's [just one outside this file](https://github.com/Automattic/jetpack/blob/733270b997e2d4a13fed55a0b5c3a1b2afb2b844/projects/packages/stats/src/class-tracking-pixel.php#L110). For `script_add_async` there's only this file and its tests, but you'd need to change this file to stop adding the filter unconditionally.

If you want to deprecate the 'async' option to this function too (which I'd recommend, it has the same naming confusion), you'd probably want to mention that deprecation in the `@since`, and mention that it's deprecated in the option's line too, and after checking that there are no uses you could use `_deprecated_argument()` to complain about it.

Reply:
thanks for the detailed instructions, I'll work on it!

Reply:
I'm also deprecating `enqueue_async_script` and updating usage.

Regarding the tests for the deprecated functions, is it ok to leave those in place?

Reply:
> As I suggested earlier, we should probably deprecate those functions.

This part should be complete.

Reply:
Yes, it's ok to leave the tests for the deprecated functions in place. They may need to be adjusted for any changes in the code though.